### ISSUE TITLE
socket.getsockname requires AI_NUMERICHOST on Python 3.7+

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#9449]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#9449|
|Type|defect|
|Created|2018-05-15 12:31:17Z|

getsockname no longer returns the interface portion of the IPv6 address, which breaks the tests and possibly the API. This adds it back and should work consistently.

<details><summary>Searchable metadata</summary>

```
trac-id__9449 9449
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_7_support Python 3.7 support
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1526387477790854 1526387477790854
changetime__1532900756029979 1532900756029979
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

PR: https://github.com/twisted/twisted/pull/1018

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 776c4d6fc9530f5119a9d0bc2f49fbb87b7fe8f2
```
#!CommitTicketReference repository="" revision="776c4d6fc9530f5119a9d0bc2f49fbb87b7fe8f2"
Merge 9449-numerichost: Fix IPv6 address changes in 3.7

Author: hawkowl
Reviewer: glyph
Fixes: ticket:9449
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

Milestone renamed
